### ISSUE TITLE
If there is a Required in oneOf, it will not work.

### ISSUE THREAD

**Issue by kahirokunn** [oazapfts/oazapfts#495]
I have this Swagger file:

```json
          "status": {
            "type": "object",
            "nullable": true,
            "properties": {
              "conditions": {
                "type": "array",
                "nullable": true,
                "items": {
                  "type": "object",
                  "oneOf": [{ "required": ["message", "status", "type"] }],
                  "properties": {
                    "message": { "type": "string" },
                    "status": {
                      "type": "string",
                      "enum": ["Unknown", "True", "False"]
                    },
                    "type": { "type": "string", "enum": ["Ready"] }
                  }
                }
              }
            }
          }
```

This outputs code like this, where the object has become any:

```ts
  status?:
    | ({
        conditions?: (any[] | null) | undefined
      } | null)
    | undefined
```

If I remove `oneOf`, it works correctly:

```json
          "status": {
            "type": "object",
            "nullable": true,
            "properties": {
              "conditions": {
                "type": "array",
                "nullable": true,
                "items": {
                  "type": "object",
                  "required": ["message", "status", "type"],
                  "properties": {
                    "message": { "type": "string" },
                    "status": {
                      "type": "string",
                      "enum": ["Unknown", "True", "False"]
                    },
                    "type": { "type": "string", "enum": ["Ready"] }
                  }
                }
              }
            }
          }
```

**Comment by kahirokunn** [comment]
@Xiphe yes

**Comment by Xiphe** [comment]
I assume then this is fixed with #503 right? Please re-open in case I misunderstood :)
